lib/nodes/CMakeLists.txt

```diff
@@ -189,6 +189,50 @@ if(WITH_NODE_WEBRTC)
     list(APPEND LIBRARIES LibDataChannel::LibDataChannel)
 endif()
 
+# Enable Delta Sharing
+if(WITH_NODE_DELTASHARING)
+    list(APPEND NODE_SRC
+        delta_sharing/delta_sharing.cpp
+        delta_sharing/delta_sharing_client.cpp
+        delta_sharing/delta_sharing_rest_client.cpp
+        delta_sharing/functions.cpp
+    )
+
+    ExternalProject_Add(restclient
+    GIT_REPOSITORY https://github.com/mrtazz/restclient-cpp
+    GIT_TAG 0.5.2
+    SOURCE_DIR restclient
+    BUILD_IN_SOURCE 1
+    UPDATE_COMMAND ""
+     PATCH_COMMAND ""
+    CONFIGURE_COMMAND ./autogen.sh && ./configure
+    BUILD_COMMAND make
+    INSTALL_COMMAND sudo make install
+    )
+
+    # Create imported target
+    add_library(restclient-cpp SHARED IMPORTED)
+    set_target_properties(restclient-cpp PROPERTIES
+        IMPORTED_LOCATION ${CMAKE_INSTALL_PREFIX}/lib/${CMAKE_SHARED_LIBRARY_PREFIX}restclient-cpp${CMAKE_SHARED_LIBRARY_SUFFIX}
+        INTERFACE_INCLUDE_DIRECTORIES ${CMAKE_INSTALL_PREFIX}/include
+    )
+
+    # Add dependency
+    add_dependencies(restclient-cpp restclient)
+
+    pkg_check_modules(ARROW REQUIRED arrow)
```

Comment:
Please the checking of dependencies to the main CMakeLists.txt as for the other node-types.